Document test database setup (#1145)
graphile/postgraphile: CONTRIBUTING.md

```

### Database setup

For testing, Postgraphile of course requires a Postgres database. [Download and install](https://www.postgresql.org/download/) it, e.g. with:
```
sudo apt-get update
sudo apt-get install postgresql postgresql-contrib postgresql-client-common
```
Then we need to set up a superuser (who can install extensions) and create test databases for the individual test suites.
You can use the [command line utilities](https://www.postgresql.org/docs/current/reference-client.html) or go with `psql`:
```
sudo -u postgres psql
  CREATE ROLE me WITH LOGIN SUPERUSER; -- PASSWORD 'mypassword'
  SET ROLE me;
  CREATE DATABASE postgraphile_test;
  COMMENT ON DATABASE postgraphile_test IS 'https://github.com/graphile/postgraphile';
  CREATE DATABASE graphileengine_test;
  COMMENT ON DATABASE graphileengine_test IS 'https://github.com/graphile/graphile-engine';
```
You may use the default `postgres` superuser as well instead of your own one.
Using your OS login for naming the user `me` requires fewer customisations down the road, as it will be the default.

The [`lds` package](./lds/README.md) does use another database `lds_test`, created by its `yarn db:init` script.

> The `pg-pubsub` package does use another database `subs`, created by its `test.sh` script.

Reviewer: ```suggestion
```

This is just a demo script, the tests themselves don't run it.

## Tests

PostGraphile uses [Jest](http://facebook.github.io/jest/) for testing to take
advantage of Jest’s snapshot feature. We test against a local database, so make sure [it is set up](#Database_setup).
PostgreSQL is by default running on `localhost:5432` (if that port isn't already used).

### Graphile Engine

Graphile Engine uses a user-configurable test database, for example `graphileengine_test`:
```
createdb graphileengine_test
```

The tests of the `postgraphile-core`, `graphile-utils` and `pg-pubsub` packages use the `TEST_DATABASE_URL` environment variable, which is mandatory.
Its format is a [connection URL](https://www.postgresql.org/docs/current/libpq-connect.html#id-1.7.3.8.3.6), passed to [node-postgres](https://node-postgres.com/features/connecting#Connection URI).

> The tests of the `lds` and `subscriptions-lds` packages use the `LDS_TEST_DATABASE_URL` environment variable, with `"lds_test"` as the fallback.

Reviewer: ```suggestion
The `lds` and `subscriptions-lds` packages' test commands will create and use a new `"lds_test"` database using the `createdb` command. If you cannot make `createdb` run directly without options (e.g. if you're not using `trust` authentication) then you may set the `LDS_TEST_DATABASE_URL` environment variable and seed the DB and run the tests manually (see the relevant `package.json` to see what is done by the `yarn test` command).
```

createdb postgraphile_test
```

> The tests use the `TEST_PG_URL` environment variable as a connection URL, which can be overwritten.

Reviewer: ```suggestion
The PostGraphile tests use the `TEST_PG_URL` environment variable as a connection URL, which can be overwritten. (This differs from the Graphile Engine `TEST_DATABASE_URL` to avoid conflicts between these two independent test suites.)
```
